nidirect_site_themes/src/Controller/SiteThemeController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\nidirect_site_themes\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Link;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Class SiteThemeController.
+ */
+class SiteThemeController extends ControllerBase {
+
+  /**
+   * Display.
+   *
+   * @return site themes tree.
+   */
+  public function display() {
+    $output = $this->printVocab('site_themes');
+    return [
+      '#type' => 'markup',
+      '#markup' => $output,
+    ];
+  }
+
+  /**
+   * Display entire tree for one vocabulary.
+   */
+  protected function printVocab($vid) {
+    $output = "<div class='item_list'><ul>";
+    $terms = $this->entityTypeManager()->getStorage('taxonomy_term')->loadTree($vid, 0, 1);
+    foreach ($terms as $term) {
+      $output .= $this->printOneLevel($vid, $term->tid);
+    }
+    $output .= "</ul></div>";
+    return $output;
+  }
+
+  /**
+   * Display one level of tree.
+   */
+  protected function printOneLevel($vid, $parent_tid) {
+    $output = "<div class='theme_item_list'><ul>";
+    $term = Term::load($parent_tid);
```

Comment:
This class has access to an injected entityTypeManager, the term should be loaded via that.

Reply:
That's fixed now too